// Copyright 2022 Google LLC
// Copyright (C) Microsoft Corporation. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package kv_server;

// Every key belongs to one namespace. One key is unique within that namespace.
// See
// https://github.com/WICG/turtledove/blob/main/FLEDGE_Key_Value_Server_API.md
// for details.
message KeyNamespace {
  enum Enum {
    KEY_NAMESPACE_UNSPECIFIED = 0;

    // For internal demo/debug.
    KV_INTERNAL = 1;

    // Used for DSP k/v servers. For DSP, all keys belong to this namespace.
    KEYS = 2;

    // Used for SSP k/v servers.
    RENDER_URLS = 3;
    AD_COMPONENT_RENDER_URLS = 4;
  }
}

// Types of files the data server recognizes as data input.
message FileType {
  enum Enum {
    FILE_TYPE_UNSPECIFIED = 0;

    DELTA = 1;

    SNAPSHOT = 2;

    LOGICAL_SHARDING_CONFIG = 3;

    // enum specific to the MICROSOFT_AD_SELECTION_BUILD. That's why not incremental.
    ANNSNAPSHOT = 99;
  }
}
